src/lib.rs

```
@@ -127,75 +140,82 @@ pub struct SevenSegment<A, B, C, D, E, F, G, Common> {
-                                             F: OutputPin,
-                                             G: OutputPin,
-                                             Common: Polarity
+impl<A, B, C, D, E, F, G, Common> SevenSegment<A, B, C, D, E, F, G, Common>
+where
+    A: OutputPin,
+    B: OutputPin,
+    C: OutputPin,
+    D: OutputPin,
+    E: OutputPin,
+    F: OutputPin,
+    G: OutputPin,
+    Common: Polarity,
 {
     /// Sets the value of the display.
     ///
     /// The valid values are 0-9. In case of invalid value, the display will be blank.
     pub fn set(&mut self, value: u8) {
         let mask = match value {
-            //    a  b  c  d  e  f  g
-            0 => (1, 1, 1, 1, 1, 1, 0),
-            1 => (0, 1, 1, 0, 0, 0, 0),
-            2 => (1, 1, 0, 1, 1, 0, 1),
-            3 => (1, 1, 1, 1, 0, 0, 1),
-            4 => (0, 1, 1, 0, 0, 1, 1),
-            5 => (1, 0, 1, 1, 0, 1, 1),
-            6 => (1, 0, 1, 1, 1, 1, 1),
-            7 => (1, 1, 1, 0, 0, 0, 0),
-            8 => (1, 1, 1, 1, 1, 1, 1),
-            9 => (1, 1, 1, 1, 0, 1, 1),
+            //      a  b  c  d  e  f  g
+            0x0 => (1, 1, 1, 1, 1, 1, 0),
+            0x1 => (0, 1, 1, 0, 0, 0, 0),
+            0x2 => (1, 1, 0, 1, 1, 0, 1),
+            0x3 => (1, 1, 1, 1, 0, 0, 1),
+            0x4 => (0, 1, 1, 0, 0, 1, 1),
+            0x5 => (1, 0, 1, 1, 0, 1, 1),
+            0x6 => (1, 0, 1, 1, 1, 1, 1),
+            0x7 => (1, 1, 1, 0, 0, 0, 0),
+            0x8 => (1, 1, 1, 1, 1, 1, 1),
+            0x9 => (1, 1, 1, 1, 0, 1, 1),
+            0xa => (1, 1, 1, 0, 1, 1, 1),
+            0xb => (0, 0, 1, 1, 1, 1, 1),
+            0xc => (1, 0, 0, 1, 1, 1, 0),
+            0xd => (0, 1, 1, 1, 1, 0, 1),
+            0xe => (1, 0, 0, 1, 1, 1, 1),
+            0xf => (1, 0, 0, 0, 1, 1, 1),
             _ => (0, 0, 0, 0, 0, 0, 0),
         };
 
         if mask.0 == Common::is_cathode() as u8 {
-            self.a.set_high();
+            let _ = self.a.set_high();
```

Please do not ignore the error, but return it using `?` operator.
Like this: 
`self.a.set_high()?;`

Same for other cases.

The signature of the function should return `Result<(), A::Error>`.